src/main/java/org/mockito/internal/framework/DefaultMockitoFramework.java

```diff
@@ -23,4 +24,24 @@ public MockitoFramework removeListener(MockitoListener listener) {
         mockingProgress().removeListener(listener);
         return this;
     }
+
+    /**
+     * Adds listener only if it is an instance of MockitoListener.
+     * Useful to avoid casting in internal code.
+     */
+    public static void tryAddListner(Object listener) {
```

Comment:
Here're my reasons:

- Multiple interfaces with single methods are easier to evolve than APIs with "big" interfaces. It makes it easier to grow / add new APIs by addition of new interfaces without breaking current compatibility.
- MockitoTestListener and MockitoListener are different. The former is used by JUnit rule to add behavior after each test that uses the rule, we cannot add this listener using Mockito.framework(). The latter is a listener for events triggered during "mocking" and is intended to be added by Mockito.framework()

Hope that helps! I'm happy to discuss better implementation. Please suggest!